### PR TITLE
feat: show server error message in push

### PR DESCRIPTION
**Pull request by alaeddine-13**
Show the error message coming from the server in push, if the code is 403

### CODE REVIEW

File: docarray/array/mixins/io/pushpull.py [hunk 1]
```diff
@@ -250,6 +250,8 @@ def _get_chunk(_batch):
         if response.ok:
             return response.json()['data']
         else:
+            if response.status_code == 403:
```
Comment (JoanFM) [inline docarray/array/mixins/io/pushpull.py:253]: 403 seems a magic number, lets use a constant

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/array/mixins/io/pushpull.py (hunk 1 above)
Comment (JoanFM) [reply docarray/array/mixins/io/pushpull.py:253]: @jina-ai/team-hubble @delgermurun @mapleeit these constants should come from hubble-sdk

Comment (mapleeit) [reply docarray/array/mixins/io/pushpull.py:253]: Does it make sense to use `http.HTTPStatus.FORBIDDEN` here? 

It's standard http status code and `libarary/http` already provides that.

File: docarray/array/mixins/io/pushpull.py [hunk 2]
```diff
@@ -250,6 +250,8 @@ def _get_chunk(_batch):
...
+                response.reason = response.json()['message']
```
Comment (delgermurun) [inline docarray/array/mixins/io/pushpull.py:254]: The `403` error is not the only one. How about trying to get a message from every error response?
All errors should have a message, but checking for its existence is also helpful to have.

File: docarray/array/mixins/io/pushpull.py (hunk 1 above)
Comment (delgermurun) [reply docarray/array/mixins/io/pushpull.py:253]: The `403` error is not the only one. How about trying to get a message from every error response?
All errors should have a message, but checking for its existence is also helpful to have.

Comment (alaeddine-13) [reply docarray/array/mixins/io/pushpull.py]: applied

Comment (mapleeit) [reply docarray/array/mixins/io/pushpull.py]: Agree with @delgermurun 

For all the status code `>= 400`, we should adopt the same strategy, which is printing the error message out.

Comment (alaeddine-13) [reply docarray/array/mixins/io/pushpull.py]: done

**Review by delgermurun — APPROVED** [review]
LGTM

File: docarray/array/mixins/io/pushpull.py [hunk 3]
```diff
@@ -250,6 +251,8 @@ def _get_chunk(_batch):
         if response.ok:
             return response.json()['data']
         else:
+            if response.status_code >= 400 and 'readableMessage' in response.json():
+                response.reason = response.json()['readableMessage']
```
Comment (JoanFM) [inline docarray/array/mixins/io/pushpull.py:255]: u should use HTTPStatus for somehing. There must be some constant from there, 400 seems magic to me

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/array/mixins/io/pushpull.py (hunk 3 above)
Comment (alaeddine-13) [reply docarray/array/mixins/io/pushpull.py:255]: I would've used it if it was an equal check (==).
I think it is more readable to use the actual number in comparison (it just means failed response).
The requests library does the same actually
image

**Review by JoanFM — APPROVED** [review]

**Pull request merged**